### PR TITLE
Add missing monaco editor preferences

### PR DESCRIPTION
**Pull request by vince-fugnitto**
Added missing `monaco-editor preferences` that could be useful for theia users.

Signed-off-by: Vincent Fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

File: packages/editor/src/browser/editor-preferences.ts [hunk 1]
```diff
@@ -309,6 +317,49 @@ export const editorPreferenceSchema: PreferenceSchema = {
...
+            'description': 'Render icons in suggestions box.'
+        },
+        'editor.showUnused': {
+            'type': 'boolean',
+            'default': true,
+            'description': 'Controls fading out of unused variables.',
+        },
+        'editor.suggestSelection': {
+            'enum': ['first', 'recentlyUsed', 'recentlyUsedByPrefix'],
+            'description': 'The history mode for suggestions',
```
Comment (elaihau) [inline packages/editor/src/browser/editor-preferences.ts]: default is missing

File: packages/editor/src/browser/editor-preferences.ts [hunk 2]
```diff
@@ -92,22 +94,22 @@ export const editorPreferenceSchema: PreferenceSchema = {
...
             'type': 'boolean',
             'default': false,
-            'description': 'Enable or disable the minimap'
+            'description': 'Enable or disable the minimap.'
         },
         'editor.minimap.showSlider': {
             'type': 'string',
             'default': 'mouseover',
-            'description': "Controls whether the minimap slider is automatically hidden. Possible values are 'always' and 'mouseover'"
+            'description': "Controls whether the minimap slider is automatically hidden. Possible values are 'always' and 'mouseover'."
```
Comment (elaihau) [inline packages/editor/src/browser/editor-preferences.ts]: if possible values are always and mouseover, why do we not make it an enum ?

File: packages/editor/src/browser/editor-preferences.ts [hunk 3]
```diff
@@ -355,7 +408,7 @@ export interface EditorConfiguration {
     'editor.roundedSelection'?: boolean
     'editor.minimap.enabled'?: boolean,
     'editor.minimap.showSlider'?: string,
-    'editor.minimap.renderCharacters'?: boolean,
+    'editor.minimap.renderCharacters'?: boolean
```
Comment (elaihau) [inline packages/editor/src/browser/editor-preferences.ts]: some props have comma at the end of line, some don't. could you make it consistent please ?

File: packages/editor/src/browser/editor-preferences.ts [hunk 4]
```diff
@@ -240,7 +282,12 @@ export const editorPreferenceSchema: PreferenceSchema = {
             'description': 'Accept suggestions on provider defined characters.'
         },
         'editor.snippetSuggestions': {
-            'enum': ['top', 'bottom', 'inline', 'none'],
+            'enum': [
+                'top',
+                'bottom',
+                'inline',
```
Comment (lmcbout) [inline packages/editor/src/browser/editor-preferences.ts]: In the other enums, yo u usually put the default value the first entry of the enum

File: packages/editor/src/browser/editor-preferences.ts [hunk 5]
```diff
@@ -267,15 +314,27 @@ export const editorPreferenceSchema: PreferenceSchema = {
...
+            'default': 'auto',
+            'description': 'Selects the folding strategy.'
+                + '\'auto\' uses the strategies contributed for the current document, \'indentation\' uses the indentation based folding strategy. '
         },
         'editor.showFoldingControls': {
-            'enum': ['always', 'mouseover'],
+            'enum': [
+                'always',
+                'mouseover'
+            ],
```
Comment (lmcbout) [inline packages/editor/src/browser/editor-preferences.ts]: Why don't we put the default as the first entry in the enum !

File: packages/editor/src/browser/editor-preferences.ts [hunk 6]
```diff
@@ -295,7 +354,12 @@ export const editorPreferenceSchema: PreferenceSchema = {
             'description': 'Enable rendering of indent guides.'
         },
         'editor.renderLineHighlight': {
-            'enum': ['none', 'gutter', 'line', 'all'],
+            'enum': [
+                'none',
+                'gutter',
+                'line',
+                'all'
+            ],
```
Comment (lmcbout) [inline packages/editor/src/browser/editor-preferences.ts]: Same here, put default first

**Pull request merged**